### ISSUE TITLE
Screen Reader: Profile - Status: Status emoji button announced without the selected emoji

### ISSUE THREAD

**Issue by elisenwazuoke** [Expensify/App#77371]
## Action Performed:
Precondition: User is logged in

1. Using VoiceOver on iOS or TalkBack on Android, open the app
2. Swipe right to focus the Account tab and double-tap to activate it
3. Swipe right to focus the Profile button and double-tap to activate it
4. Swipe right to focus the Status menu item and double-tap to activate it
5. Swipe right to focus the Status Emoji button and note if the currently selected emoji is announced

## Expected Result:
Screen readers should announce the button including the currently selected emoji, for example:
“Status emoji, smiling face selected, button”
Screen reader users should be informed what emoji is currently applied to their status.

## Actual Result:
The control is announced only as "Status emoji, button".
The selected emoji is not included in the announcement so screen reader users cannot tell which emoji is currently applied.

## Area issue was found in:
Profile > Status

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Expose the selected emoji via the accessibilityLabel to ensure it is announced, e.g.:
- 'Status Emoji, Rocket emoji selected'
https://appt.org/en/docs/react-native/samples/accessibility-label
https://reactnative.dev/docs/accessibility#accessibilitylabel

## Environment(s)
Apple iPhone 14 Pro - iOS 18.7.1, Wi-Fi - Native (No mobile browser)
Samsung Galaxy Note10 Plus - Android 12, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos
<img width="1179" height="2441" alt="Image" src="https://github.com/user-attachments/assets/993f4eb3-2211-4297-9e88-1f7035e6bebc" />

https://github.com/user-attachments/assets/4575ef0d-0f41-415d-96bc-a578ca80e768

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7006140
**Issue reported by:** NA
**Slack conversation:** NA
